tests/integration/ambient/wasm_test.go

```diff
@@ -0,0 +1,251 @@
+	args := map[string]any{
+		"WasmPluginName": pluginName,
+	}
+	if err := ctx.ConfigIstio().EvalFile(apps.Namespace.Name(), args, path).Delete(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func sendTraffic(ctx framework.TestContext, checker echo.Checker, options ...retry.Option) {
+	ctx.Helper()
+	if len(GetClientInstances()) == 0 {
+		ctx.Fatal("there is no client")
+	}
+	cltInstance := GetClientInstances()[0]
+
+	defaultOptions := []retry.Option{retry.Delay(100 * time.Millisecond), retry.Timeout(200 * time.Second)}
+	httpOpts := echo.CallOptions{
+		To: GetTarget(),
+		Port: echo.Port{
+			Name: "http",
+		},
+		HTTP: echo.HTTP{
+			Path:   "/path",
+			Method: "GET",
+		},
+		Count: 1,
+		Retry: echo.Retry{
+			Options: append(defaultOptions, options...),
+		},
+		Check: checker,
+	}
+
+	_ = cltInstance.CallOrFail(ctx, httpOpts)
+}
+
+func sendTrafficToHostname(ctx framework.TestContext, checker echo.Checker, hostname string, options ...retry.Option) {
+	ctx.Helper()
+	if len(GetClientInstances()) == 0 {
+		ctx.Fatal("there is no client")
+	}
+	cltInstance := GetClientInstances()[0]
+
+	defaultOptions := []retry.Option{retry.Delay(100 * time.Millisecond), retry.Timeout(200 * time.Second)}
+	httpOpts := echo.CallOptions{
+		Address: hostname,
+		Port: echo.Port{
+			Name:        "http",
+			ServicePort: 80,
+			Protocol:    protocol.HTTP,
+		},
+		HTTP: echo.HTTP{
+			Path:    "/path",
+			Method:  "GET",
+			Headers: headers.New().WithHost(fmt.Sprintf("%s.com", GetTarget().ServiceName())).Build(),
+		},
+		Count: 1,
+		Retry: echo.Retry{
+			Options: append(defaultOptions, options...),
+		},
+		Check: checker,
+	}
+
+	_ = cltInstance.CallOrFail(ctx, httpOpts)
+}
+
+func getTargetRefValues(targetType, targetName string) (kind, group, name string) {
+	switch targetType {
+	case "gateway":
+		return "Gateway", "gateway.networking.k8s.io", targetName
+	case "service":
+		return "Service", "\"\"", targetName
+	default:
+		return "", "", ""
+	}
+}
+
+func TestWasmPluginConfigurations(t *testing.T) {
+	framework.NewTest(t).
+		Run(func(t framework.TestContext) {
+			testCases := []struct {
+				desc         string
+				name         string
+				testHostname string
+				targetType   string
+				targetName   string
+			}{
+				{
+					desc:         "Configure WebAssembly filter for gateway",
+					name:         "gateway-wasm-test",
+					testHostname: fmt.Sprintf("%s-gateway-istio.%s.svc.cluster.local", GetTarget().ServiceName(), apps.Namespace.Name()),
+					targetType:   "gateway",
+					targetName:   fmt.Sprintf("%s-gateway", GetTarget().(echo.Instances).ServiceName()),
+				},
+				{
+					desc:       "Configure WebAssembly filter for waypoint",
+					name:       "waypoint-wasm-test",
+					targetType: "gateway",
+					targetName: constants.DefaultNamespaceWaypoint,
+				},
+				{
+					desc:       "Configure WebAssembly filter for specific service",
+					name:       "service-wasm-test",
```

Comment:
this test has to be skipped right? See if you can leverage t.Skip() functionality

Reply:
```
			for _, tc := range testCases {
				if tc.name == "service-wasm-test" {
					t.Skip("skipping service-wasm-test based on the issue https://github.com/istio/istio/issues/51288/")
```

I did like this. Is this okay ?